core/workflow-operator/src/main/scala/edu/uci/ics/amber/operator/visualization/radarPlot/RadarPlotOpDesc.scala

```diff
@@ -0,0 +1,167 @@
+
+  @JsonProperty(value = "traceNameAttribute", required = false)
+  @JsonSchemaTitle("Trace Name Column")
+  @JsonPropertyDescription("Optional column to use for naming each radar trace")
+  @AutofillAttributeName
+  var traceNameAttribute: String = ""
+
+  @JsonProperty(value = "normalize", required = false)
+  @JsonSchemaTitle("Normalize Data")
+  @JsonPropertyDescription("Normalize the radar plot values")
+  var normalize: Boolean = true
+
+  override def getOutputSchemas(
+      inputSchemas: Map[PortIdentity, Schema]
+  ): Map[PortIdentity, Schema] = {
+    val outputSchema = Schema()
+      .add("html-content", AttributeType.STRING)
+    Map(operatorInfo.outputPorts.head.id -> outputSchema)
+    Map(operatorInfo.outputPorts.head.id -> outputSchema)
+  }
+
+  override def operatorInfo: OperatorInfo =
+    OperatorInfo(
+      "Radar Plot",
+      "View the result in radar plot",
+      OperatorGroupConstants.VISUALIZATION_BASIC_GROUP,
+      inputPorts = List(InputPort()),
+      outputPorts = List(OutputPort(mode = OutputMode.SINGLE_SNAPSHOT))
+    )
+
+  def generateRadarPlotCode(): String = {
+    assert(selectedAttributes.nonEmpty)
+
+    val attrList = selectedAttributes.map(attr => s""""$attr"""").mkString(", ")
+    val traceNameCol = traceNameAttribute match {
+      case null | "" => "None"
+      case col       => s"'$col'"
+    }
+    val normalizePython = if (normalize) "True" else "False"
+
+    s"""
+       |        categories = [$attrList]
+       |        trace_name_col = $traceNameCol
+       |        normalize = $normalizePython
+       |        max_vals = {attr: float('-inf') for attr in categories}
+       |
+       |        fig = go.Figure()
+       |
+       |        if normalize:
+       |            for _, row in table.iterrows():
+       |                for attr in categories:
+       |                    max_vals[attr] = max(max_vals[attr], row[attr])
+       |
+       |        for _, row in table.iterrows():
+       |            trace_name = row[trace_name_col] if trace_name_col is not None else ""
+       |            if normalize:
+       |                original_vals = []
+       |                normalized_vals = []
+       |                for attr in categories:
+       |                    original_vals.append(f"{attr}: {row[attr]}")
+       |                    normalized_vals.append(
+       |                        row[attr] / max_vals[attr] if max_vals[attr] != 0 else 0
+       |                    )
+       |
+       |                fig.add_trace(go.Scatterpolar(
+       |                    r=normalized_vals,
+       |                    theta=categories,
+       |                    fill='toself',
+       |                    name=str(trace_name) if trace_name is not None else "",
+       |                    text=original_vals,
+       |                    hoverinfo="text"
+       |                ))
+       |            else:
+       |                fig.add_trace(go.Scatterpolar(
```

Comment:
The `if normalize:` block introduces duplication. Both branches repeat almost the same logic.